recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,260 @@
+        self.requires("zlib/1.2.11")
+        if self.settings.os != "Windows":
+            self.requires("libcurl/7.74.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+        # Keeping these unused folders make the source copy fail on Windows
+        tools.rmdir(os.path.join(self._source_subfolder, "code-generation", "generator"))
+        tools.rmdir(os.path.join(self._source_subfolder, "aws-cpp-sdk-core-tests"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        build_only = ["core"]
+        for sdk in self.sdks:
+            if getattr(self.options, sdk):
+                build_only.append(sdk)
+        self._cmake.definitions["BUILD_ONLY"] = ";".join(build_only)
+
+        self._cmake.definitions["BUILD_DEPS"] = False
+        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
+
+        self._cmake.definitions["MINIMIZE_SIZE"] = self.options.min_size
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in self.settings.compiler.runtime
+
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.components["core"].libs = ["aws-cpp-sdk-core"]
+        self.cpp_info.components["core"].requires = [
+                "aws-c-common::aws-c-common-lib",
+                "aws-c-event-stream::aws-c-event-stream-lib",
+                "zlib::zlib",
+                ]
+
+        for sdk in self.sdks:
+            if getattr(self.options, sdk):
+                self.cpp_info.components[sdk].libs = ["aws-cpp-sdk-" + sdk]
+
+        if self.settings.os == "Windows":
+            self.cpp_info.components["core"].system_libs.extend(["winhttp", "wininet", "bcrypt", "userenv", "version", "ws2_32"])
+        else:
+            self.cpp_info.components["core"].requires.append("libcurl::curl")
+
+        if self.settings.os == "Linux":
+            self.cpp_info.components["core"].system_libs.append("atomic")
+
+        lib_stdcpp = tools.stdcpp_library(self)
+        if lib_stdcpp:
+            self.cpp_info.components["core"].system_libs.append(lib_stdcpp)
```

Comment:
Few comments:

- these conan components names are too generic for pkgconfig generator

- config file name is `aws-cpp-sdk-core-config.cmake` (is there one config file per component by the way?)

  ```
  -- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/aws-cpp-sdk-core/aws-cpp-sdk-core-config.cmake
  -- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/aws-cpp-sdk-core/aws-cpp-sdk-core-config-version.cmake
  ```

- What are the names of imported targets?

- Among these installed CMake module files, anything to keep and add to build_modules?

```
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/AWSSDKConfigVersion.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platformDeps.cmake
-- Up-to-date: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/resolve_platform.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/AWSSDKConfig.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/sdksCommon.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/sdks.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/utilities.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/initialize_project_version.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform/windows.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform/custom.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform/android.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform/linux.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform/apple.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/platform/unix.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/make_uninstall.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/external_dependencies.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/dependencies.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/build_external.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/setup_cmake_find_module.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-sdk-cpp/1.8.130/_/_/package/ceeb5fa96a6bc55fa9df1c94b3e5f16e86db4b0d/lib/cmake/AWSSDK/compiler_settings.cmake
```